rplugin/python3/deoplete/sources/beancount.py

```diff
@@ -46,9 +46,16 @@ def gather_candidates(self, context):
         if re.match(r'^(\s)+\w+$', context['input']):
             return [{'word': x, 'kind': 'account'} for x in attrs['accounts']]
         # directive followed by account
-        if re.match(r'(balance|document|note|open|close|pad(\s\S*)?)\s\w+$',
-                    context['input']):
+        if re.search(r'(balance|document|note|open|close|pad(\s\S*)?)\s\w+$',
+                     context['input']):
             return [{'word': x, 'kind': 'account'} for x in attrs['accounts']]
+        # commodity after number
+        if re.search(r'([0-9]+|[0-9][0-9,]+[0-9])(\.[0-9]*)?\s\w+',
```

Comment:
Thanks, yes all the regexes should probably end in a `$`.

Will also add a whitespace at the front, numbers should always be preceded by a space.